### PR TITLE
[REEF-728] Complete implementation of ICompletedTask

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Piping the completed task message from Java to C#.

JIRA:
  [REEF-728](https://issues.apache.org/jira/browse/REEF-728)

### CODE REVIEW

**Comment by jwang98052** [comment]
I will have a look.

**Comment by jwang98052** [comment]
Looks good. 

I noticed you have updated SimpleEvnetHandler, tah tis good. If you can add a handler which implements  IObserver<ICompletedTask> and add set it to REEF.Driver.DriverConfiguration.OnTaskCompleted in AllHandlers, that would make the example more complete.

File: lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/HelloSimpleEventHandlers.cs [hunk 1]
```diff
@@ -174,7 +174,7 @@ public void OnNext(ICompletedTask value)
         {
             using (Logger.LogFunction("HelloSimpleEventHandlers::CompletedTask received"))
             {
-                Logger.Log(Level.Info, string.Format(CultureInfo.InvariantCulture, "Received CompletedTask: {0}, task id: {1}", value.Id, _taskContext.CurrentTaskId()));
+                Logger.Log(Level.Info, string.Format(CultureInfo.InvariantCulture, "Received CompletedTask: {0}, task id: {1}, task value: {2}.", value.Id, _taskContext.CurrentTaskId(), ByteUtilities.ByteArrarysToString(value.Message)));
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/HelloSimpleEventHandlers.cs]: Logging the message into the `Info` level seems too verbose. Maybe have a second or different log line for the `Verbose` level?

File: lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/CompletedTaskBridge.java [hunk 2]
```diff
@@ -19,14 +19,15 @@
 package org.apache.reef.javabridge;
 
 import org.apache.reef.driver.task.CompletedTask;
+import org.apache.reef.io.Message;
 
-public class CompletedTaskBridge extends NativeBridge {
+public class CompletedTaskBridge extends NativeBridge implements Message {
```
Comment (markusweimer) [inline lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/CompletedTaskBridge.java]: Make the class `final` and add an `@Private` while we touch it.

File: lang/cs/Org.Apache.REEF.Driver/Bridge/Clr2java/IClr2JavaTaskMessage.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+namespace Org.Apache.REEF.Driver.Bridge.Clr2java
+{
+    public interface IClr2JavaTaskMessage
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Driver/Bridge/Clr2java/IClr2JavaTaskMessage.cs]: Please add documentation of this interface

**Comment by markusweimer** [comment]
I've done a pass as well.

**Comment by afchung** [comment]
@markusweimer @jwang98052 Addressed review comments, please have another look. Thanks!

**Comment by afchung** [comment]
@jwang98052 Addressed comments. Good catch, I always forget that VS only adds the file after closing the process.

**Comment by jwang98052** [comment]
I will test and merge

**Comment by jwang98052** [comment]
Test RunSimpleEventHandlerOnLocalRuntime failed at:

EXIT: 9/29/2015 8:36:27 PM ClrSystemHandlerWrapper::Call_ClrSystemCompletedTask_OnNext. Duration: [00:00:00.0041098].
<C++>InteropUtil Error: 0 : 2015-09-29T20:36:27.2190837-07:00 0003
ERROR: Exception in Call_ClrSystemCompletedTask_OnNextencountered error [System.ArgumentNullException: Array cannot be null.
Parameter name: bytes
   at System.Text.Encoding.GetString(Byte[] bytes)
   at Org.Apache.REEF.Utilities.ByteUtilities.ByteArrarysToString(Byte[] b)
   at Org.Apache.REEF.Tests.Functional.Bridge.HelloSimpleEventHandlers.OnNext(ICompletedTask value)
   at Org.Apache.REEF.Driver.Bridge.ClrSystemHandler`1.OnNext(T value)
   at Org.Apache.REEF.Driver.Bridge.ClrSystemHandlerWrapper.Call_ClrSystemCompletedTask_OnNext(UInt64 handle, ICompletedTaskClr2Java clr2Java)
   at Java_org_apache_reef_javabridge_NativeInterop_clrSystemCompletedTaskHandlerOnNext(JNIEnv_* env, _jclass* cls, Int64 handler, _jobject* jcompletedTask, _jobject* jlogger)] with mesage [Array cannot be null.
Parameter name: bytes] and stack trace [   at System.Text.Encoding.GetString(Byte[] bytes)
   at Org.Apache.REEF.Utilities.ByteUtilities.ByteArrarysToString(Byte[] b)
   at Org.Apache.REEF.Tests.Functional.Bridge.HelloSimpleEventHandlers.OnNext(ICompletedTask value)
   at Org.Apache.REEF.Driver.Bridge.ClrSystemHandler`1.OnNext(T value)
   at Org.Apache.REEF.Driver.Bridge.ClrSystemHandlerWrapper.Call_ClrSystemCompletedTask_OnNext(UInt64 handle, ICompletedTaskClr2Java clr2Java)
   at Java_org_apache_reef_javabridge_NativeInterop_clrSystemCompletedTaskHandlerOnNext(JNIEnv_* env, _jclass* cls, Int64 handler, _jobject* jcompletedTask, _jobject* jlogger)]

**Comment by afchung** [comment]
@jwang98052 Sorry about that. Accounted for the null case now. Thanks!

**Pull request closed without merging**